code/modules/mob/living/carbon/alien/larva/larva.dm

```diff
@@ -24,57 +25,60 @@
 	regenerate_icons()
 	add_language("Xenomorph")
 	add_language("Hivemind")
+	AddSpell(new /obj/effect/proc_holder/spell/alien_spell/evolve_larva)
 
 /mob/living/carbon/alien/larva/get_caste_organs()
 	. = ..()
-	. += /obj/item/organ/internal/xenos/plasmavessel/larva
+	. += /obj/item/organ/internal/alien/plasmavessel/larva
 
 
-//This needs to be fixed
 /mob/living/carbon/alien/larva/Stat()
 	..()
 	stat(null, "Progress: [amount_grown]/[max_grown]")
 
-/mob/living/carbon/alien/larva/adjustPlasma(amount)
+/mob/living/carbon/alien/larva/add_plasma(amount)
 	if(stat != DEAD && amount > 0)
 		amount_grown = min(amount_grown + 1, max_grown)
 	..(amount)
 
+/mob/living/carbon/alien/larva/proc/cast_larva()
```

Comment:
This shouldn't need to exist if the transfer spell is a click spell

Reply:
See my reasoning on making these touch spells, just felt terrible in testing

Reply:
Removed entirely